Fix active state of Collections in navigation with Content Versions and Bookmarks

paescuj opened the pull request:
## Scope

- Fix active state of collection in navigation for Content Versions and Bookmarks
-  Don't add bookmark to route when navigating back to "parent" 

### Before

<img width="638" src="https://github.com/directus/directus/assets/5363448/7dcbfd63-b718-4d50-987f-042ea0cad41d">

### After

<img width="638" src="https://github.com/directus/directus/assets/5363448/e507b1fd-666c-4a2e-bce5-4c3407c47fc0">

## Potential Risks / Drawbacks

I don't see any reason to require query match for these Collection navigation items under Content. I think we always want to match the Collection on all sub-routes (incl. queries).
But if there should be a reason I'm missing, it could display undesired active states.

## Review Notes / Questions

See above

paescuj committed:
Fix active state nav item under Content with query in URL

paescuj committed:
Add changeset

licitdev reviewed (commented):
The removal of `query` fixes the active status of the collection link when testing on collections without bookmarks. However, the issue persists if the collection has bookmarks. 🤔 

<img width="627" alt="image" src="https://github.com/directus/directus/assets/26413686/bca56800-a171-4876-899c-d02d8e99cef7">

paescuj committed:
Merge branch 'main' into fix-content-nav-item-active

paescuj committed:
Fix TS issue

paescuj committed:
Don't add bookmark to route when navigating to "parent"

paescuj committed:
Enhance active item state

- Only match specified queries (e.g. still a match when link is
  '?bookmark=1' but active route is '?bookmark=1&version=draft')
  - If link has no queries, it is a non-match when active route has some
- Make parent active when menu is collapsed but bookmark still selected

licitdev reviewed (approved):
LGTM! Tested the fixes to work well~ 👍
